python/ray/workflow/common.py

```diff
@@ -79,6 +79,17 @@ class StepType(str, Enum):
     READONLY_ACTOR_METHOD = "READONLY_ACTOR_METHOD"
 
 
+CheckpointModeType = Union[None, bool]
+
+
+@unique
+class CheckpointMode(Enum):
+    """All checkpoint modes."""
+    DEFAULT = None
```

Comment:
please add some brief explanation about each one of them, especially what does default here mean